ci(boilerplate): introduce modern front-end development toolchain

sabertazimi opened the pull request.

killerducky commented:
I don't use node, I just use vscode live server. I added a readme to explain that and a few other things mostly aimed at translators. 

What's the advantage of doing node deployment?

sabertaz commented:
I think python toolchain is good enough. For most front-end developers (e.g. me), Node.js environment and npm package manager are standard things for compilng, linting, bundling and building etc.That's the only reason I introduce [vite](https://vitejs.dev/guide/why.html) toolchain.If you prefer python toolchain, I will close this PR.

sabertazimi committed:
ci(boilerplate): introduce modern front-end development toolchain

sabertazimi committed:
Merge branch 'master' into ci/boilerplate

sabertazimi committed:
fix(version query): remove redundant version query

After running `npm run build`,
Vite will produce minimized stylesheets and scripts
with content-hash filename,
e.g `dist/assets/index-BFVpicZP.css   6.69 kB │ gzip: 1.77 kB`,
`dist/assets/index-yJXb5GAI.js   24.02 kB │ gzip: 7.28 kB`.
Vite will also change reference in `index.html` automatically.

So there is no need to add version query mannually.

sabertazimi commented:
After running `npm run build`,
Vite will produce minimized stylesheets and scripts
with content-hash filename,
e.g `dist/assets/index-BFVpicZP.css   6.69 kB │ gzip: 1.77 kB`,
`dist/assets/index-yJXb5GAI.js   24.02 kB │ gzip: 7.28 kB`.
Vite will also change reference in `index.html` automatically.

So there is no need to add version query mannually.
